File: aries-backchannels/dotnet/server/Controllers/DidController.cs
using Microsoft.AspNetCore.Mvc;
using System.Threading.Tasks;

using Hyperledger.Aries.Configuration;
using Hyperledger.Aries.Agents;

namespace DotNet.Backchannel.Controllers
{
    [Route("agent/command/did")]
    [ApiController]
    public class DidController : ControllerBase
    {
        private readonly IAgentProvider _agentContextProvider;
        private readonly IProvisioningService _provisionService;

        public DidController(
            IAgentProvider agentContextProvider,
            IProvisioningService provisionService)
        {
            _agentContextProvider = agentContextProvider;
            _provisionService = provisionService;
        }


        [HttpGet]
        public async Task<IActionResult> GetPublicDid()
        {
            var context = await _agentContextProvider.GetContextAsync();
            var issuer = await _provisionService.GetProvisioningAsync(context.Wallet);

            return Ok(new
            {
                did = issuer.IssuerDid,
                verkey = issuer.IssuerVerkey
            });
        }
    }
}

File: aries-backchannels/javascript/server/src/controllers/DidController.ts
import { Controller, Get } from '@tsed/common'
import { BaseController } from '../BaseController'
import { TestHarnessConfig } from '../TestHarnessConfig'

@Controller('/agent/command/did')
export class DidController extends BaseController {
  public constructor(testHarnessConfig: TestHarnessConfig) {
    super(testHarnessConfig)
  }

  @Get()
  getPublicDid() {
    const didInfo = this.agent.publicDid

    return didInfo
  }
}
